flang/lib/Lower/Bridge.cpp

```diff
@@ -499,6 +504,101 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     }
   }
 
+  mlir::Value createThreadprivateVarClone(
+      const Fortran::semantics::Symbol &sym) override final {
+    auto loc = genLocation(sym.name());
+    Fortran::lower::SymbolBox sb = lookupSymbol(sym);
+    fir::ExtendedValue exv = sb.toExtendedValue();
+
+    auto allocate = [&](mlir::Type ty) -> mlir::Value {
+      return builder->allocateLocal(
+          loc, ty, mangleName(sym), toStringRef(sym.GetUltimate().name()),
+          /*pinned=*/true, fir::factory::getExtents(*builder, loc, exv),
+          fir::getTypeParams(exv),
+          sym.GetUltimate().attrs().test(Fortran::semantics::Attr::TARGET));
+    };
+
+    mlir::Value allocaVal = nullptr;
+    if (const auto *commonDet =
+            sym.detailsIf<Fortran::semantics::CommonBlockDetails>()) {
+      mlir::Type symType = getSymbolAddress(sym).getType();
+      auto refTy = symType.dyn_cast<fir::ReferenceType>();
+      symType = refTy.getEleTy();
+      allocaVal = allocate(symType);
+
+      for (const auto &mem : commonDet->objects()) {
+        const auto *memDet =
+            mem->detailsIf<Fortran::semantics::ObjectEntityDetails>();
+        assert(memDet != nullptr && "No common member object entity found");
+        Fortran::lower::SymbolBox memCsb = lookupSymbol(*mem);
+        auto *cvtSymOp = memCsb.getAddr().getDefiningOp();
+        auto *coordOp = cvtSymOp->getOperand(0).getDefiningOp();
+        auto cvtBlkVal = builder->createConvert(
+            loc, coordOp->getOperand(0).getType(), allocaVal);
+        auto coordVal = builder->create<fir::CoordinateOp>(
+            loc, cvtSymOp->getOperand(0).getType(), cvtBlkVal,
+            coordOp->getOperand(1));
+        auto cvtSymVal =
+            builder->createConvert(loc, memCsb.getAddr().getType(), coordVal);
+
+        fir::ExtendedValue memCexv = memCsb.toExtendedValue();
+        auto memExval = memCexv.match(
```

Comment:
Why can't `fir::substBase` be used here too (I see it is used below).

Reply:
Good point. Will fix it.